src/batch.rs

```diff
@@ -291,10 +291,17 @@ impl<'a> BatchBuilder<'a> {
 
     pub fn push_complex_clip(&mut self, clip: &[ComplexClipRegion]) {
         // TODO(gw): Handle nested complex clips!
-        if clip.len() > 0 {
-            self.complex_clip = Some(clip[0]);
-        } else {
-            self.complex_clip = None;
+        self.complex_clip = match clip.len() {
+            0 => None,
+            n if n > 1 => {
+                let internal_clip = clip.last().unwrap();
+                if clip[..clip.len()].iter().all(|current_clip| current_clip.contains_may_be_false_negative(internal_clip)) {
+                    Some(*internal_clip)
+                } else {
+                    Some(clip[0])
+                }
+            },
+            _ => Some(clip[0]),
```

Comment:
Could put this one before the above match and say `1 => Some(clip[0])`, then remove the guard